rholang/src/test/scala/coop/rchain/rholang/interpreter/ReduceSpec.scala

```diff
@@ -894,4 +894,50 @@ class ReduceSpec extends FlatSpec with Matchers with PersistentStoreTester {
       )
     )
   }
+
+
+  "1 matches 1" should "return true" in {
+    implicit val errorLog = new Runtime.ErrorLog()
+    val result = withTestSpace { space =>
+      implicit val env = Env.makeEnv[Par]()
+      val reducer      = RholangOnlyDispatcher.create[Task, Task.Par](space).reducer
+
+      val inspectTask =  reducer.evalExpr(EMatches(GInt(1), GInt(1)))
+
+      Await.result(inspectTask.runAsync, 3.seconds)
+    }
+
+    result.exprs should be (Seq(Expr(GBool(true))))
+    errorLog.readAndClearErrorVector should be(Vector.empty[InterpreterError])
+  }
+
+  "1 matches 0" should "return false" in {
+    implicit val errorLog = new Runtime.ErrorLog()
+    val result = withTestSpace { space =>
+      implicit val env = Env.makeEnv[Par]()
+      val reducer      = RholangOnlyDispatcher.create[Task, Task.Par](space).reducer
+
+      val inspectTask =  reducer.evalExpr(EMatches(GInt(1), GInt(0)))
+
+      Await.result(inspectTask.runAsync, 3.seconds)
+    }
+
+    result.exprs should be (Seq(Expr(GBool(false))))
+    errorLog.readAndClearErrorVector should be(Vector.empty[InterpreterError])
+  }
```

Comment:
These 2 examples are kind of trivial. It would be nice to see a more complicated example.

Reply:
The matching functionality is already tested with the tests for the `match-case` construction. 

What I'm testing here is that the matching algorithm is called when a `matches` expression is reduced.

Since I haven't done any changes to the matching functionality, adding more ReduceSpec tests for the `matches` operator doesn't improve the test coverage.

Reply:
The examples in 84d3b59 should cover the usage of BoundevVars and Connectives